src/System.IO.FileSystem/tests/File/GetSetTimes.cs

```diff
@@ -34,6 +34,26 @@ public void BirthTimeIsNotNewerThanLowestOfAccessModifiedTimes()
             Assert.InRange(File.GetCreationTimeUtc(path), DateTime.MinValue, File.GetLastWriteTimeUtc(path));
         }
 
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Linux)]
+        public void CreationTimeSet_GetReturnsExpected_WhenNotInFuture()
+        {
+            // On Linux, we synthesize CreationTime from the oldest of status changed time (ctime) and write time (mtime).
+            // Changing the CreationTime, updates mtime and causes ctime to change to the current time.
+            // When setting CreationTime to a value that isn't in the future, getting the CreationTime should return the same value.
+
```

Comment:
We generally avoid Path.GetTempFileName in favor of using FileCleanupTestBase, which will ensure the file is deleted.

Reply:
Btw I believe GetTempFileName actually creates the file..

Reply:
confirm: GetTempFileName creates file with length 0 - in my personal tests for this i have used a File.Delete after getTempfilename to be sure that i have full control over file creation....